### ISSUE TITLE
Rewrite READMEs and the INSTALL in Markdown

### ISSUE THREAD

**Issue by mahrud** [Macaulay2/M2#1244]
Given that most people get Macaulay2 from Github (excluding users who download a distributed package), it's worth the effort to rewrite the READMEs and the INSTALL guide in Markdown and possibly break pieces into separate documents. I've already added [INSTALL-CMake.md](https://github.com/Macaulay2/M2/blob/master/M2/INSTALL-CMake.md).

Here are various suggestions based on a standard open source project hosted on Github:
- Preamble
  - funding acknowledgement and brief copyright info should be at the end of the top level README.md
  - the full license should be in the top level LICENSE.md, so Github can know the license
- Getting the source code
  -  should link to the right wiki page for more details
- Programs
  - probably should be removed
- Libraries
  - each distribution, or at least platform (especially Mac) should be in a separate guide
  - Cygwin should be removed and the WSL2 guide should be added in a separate guide
  - Docker build can be listed as well
- Compiling GCC
  - probably should be removed, or at best moved to separate doc
- Choosing a place for the repository
  - probably should be removed
- Building a fat source tarfile to avoid further downloads
  - doesn't seem necessary to mention, but can be noted on the "Getting the source" section
- Compiling Macaulay2 (finally!)
- Compiling for multiple machine architectures
- HTML validation
  - probably should be in the README of the appropriate folder
- Rerunning the package examples
  - this section is very outdated and probably should be a README in the packages directory
  - phcpack link is broken
  - convex.m link is broken
  - the others are outdated or unnecessary (e.g. links to gmp and mpfr?)
- Warning messages
  - probably should be removed (useful section, except the only warning mentioned can be ignored)
- Autoconf, autoreconf, libtool, ...
- For distributors
- Unexpected problems
  - probably should be removed

Other files that should be added:
- links to the wiki, Macaulay2 website, documentation, etc.
- a proper CONTRIBUTING.md guide (see for instance [Julia's contribution guide](https://github.com/JuliaLang/julia/blob/master/CONTRIBUTING.md))
- a CITATION.md file
- a CHANGELOG (optional, but useful for packaging)

**Comment by mahrud** [comment]
@DanGrayson the INSTALL file is getting more outdated. INSTALL.mac could also be removed, since the brew package is now available, or perhaps moved to `BUILD/dan`, in case anybody is interested in your build script.

**Comment by DanGrayson** [comment]
> @DanGrayson the INSTALL file is getting more outdated. INSTALL.mac could also be removed, since the brew package is now available, or perhaps moved to `BUILD/dan`, in case anybody is interested in your build script.

What needs updating in the INSTALL file?

I think M2/INSTALL.mac is a good location for it.

**Comment by mahrud** [comment]
> What needs updating in the INSTALL file?

Did you read the whole laundry list in the first message?

**Comment by DanGrayson** [comment]
Yes, long ago, but it doesn't identify any particular item that is out of date.

I'm against depending on the wiki for instructions -- the source code repository should contain complete instructions.

**Comment by mahrud** [comment]
Then read it again, this time not dismissively. Here is one:
> Cygwin should be removed and the WSL2 guide should be added in a separate guide
